Userland/Libraries/LibGfx/Font/BitmapFont.cpp

```diff
@@ -233,9 +233,14 @@ RefPtr<BitmapFont> BitmapFont::load_from_file(DeprecatedString const& path)
 
 ErrorOr<NonnullRefPtr<BitmapFont>> BitmapFont::try_load_from_file(DeprecatedString const& path)
 {
-    auto file = TRY(Core::MappedFile::map(path));
-    auto font = TRY(load_from_memory((u8 const*)file->data()));
-    font->m_mapped_file = file;
+    auto mapped_file = TRY(Core::MappedFile::map(path));
+    return try_load_from_mapped_file(mapped_file);
+}
+
+ErrorOr<NonnullRefPtr<BitmapFont>> BitmapFont::try_load_from_mapped_file(Core::MappedFile const& mapped_file)
```

Comment:
This function should take a `RefPtr<Core::MappedFile>` by value and move() it into place to make it clear to the caller that we are going to take shared ownership of the object. Taking a RefPtr from a const& to an object is bad juju and probably shouldn't work. `RefPtr</* non-const */ T>` constructed from `T const&` is no bueno. Andreas had a change a few weeks ago to enforce this, but it broke Jakt so we rolled it back.

To make that flow better, `try_load_from_file` should move() the mapped file it creates into this function.